source/promap.js

```diff
@@ -0,0 +1,13 @@
+import _curry3 from './internal/_curry3';
+import pipe from './pipe';
+
+
+var promap = _curry3(function promap(f, g, profunctor) {
+  return typeof profunctor['fantasy-land/promap'] === 'function'
+    ? profunctor['fantasy-land/promap'](f, g)
+    : typeof profunctor.promap === 'function'
+      ? profunctor.promap(f, g)
+      : pipe(f, profunctor, g);
```

Comment:
I would prefer not to use `pipe` in the implementation.

Reply:
In favor of `compose` or a lambda?

Reply:
A lambda.  Although we have some helper functions that get used internally, we tend to write our internals with as little dependency as we can upon the public API.  The idea is to use as little of the end-user's performance budget as we can in the library code.